Eatery/Controllers/Eateries/Campus/CampusEateryMealTableViewController.swift

```diff
@@ -88,80 +106,90 @@ class CampusEateryMealTableViewController: UITableViewController {
         ) as! MealStationTableViewCell
 
         cell.titleLabel.text = "No menu available"
-        cell.contentLabel.attributedText = NSAttributedString(string: "")
 
         return cell
     }
 
     /// Create a table view cell when there is a single dining station in the menu
     private func menuItemCell(in tableView: UITableView, forRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let menu = menu, let station = menu.data.first else {
+        guard let menuData = menu?.data else {
             return emptyMenuCell(in: tableView, forRowAt: indexPath)
         }
-
+        guard let menu = menu else {
+            return emptyMenuCell(in: tableView, forRowAt: indexPath)
+        }
+        let station = menu.data[menuData.index(menuData.startIndex, offsetBy: indexPath.section)]
```

Comment:
I'm not a fan of indexing into a dictionary. Swift dictionaries are not guaranteed to be in any specific order. If order matters, convert the dictionary into an array. You say `Sort.sortMenu(menu.data.map { ($0, $1) })` three times, suggesting that order does matter. Instead of sorting multiple times (which can be expensive), why not sort it once, store it in the class, and index directly into the sorted array? There appears to be no point of keeping it as a dictionary because you never actually index into the dictionary by key as far as I can tell.